sys/include/net/ieee802154/radio.h

```diff
@@ -0,0 +1,1052 @@
+     * @brief the device reports when the transmission is done
+     */
+    IEEE802154_CAP_IRQ_TX_DONE,
+    /**
+     * @brief the device reports the start of a frame (SFD) when received.
+     */
+    IEEE802154_CAP_IRQ_RX_START,
+    /**
+     * @brief the device reports the end of the CCA procedure
+     */
+    IEEE802154_CAP_IRQ_CCA_DONE,
+    /**
+     * @brief the device provides the number of retransmissions
+     */
+    IEEE802154_CAP_FRAME_RETRANS_INFO,
+} ieee802154_rf_caps_t;
+
+/**
+ * @brief IEEE802.15.4 transceiver states (not to confuse with device states)
+ */
+typedef enum {
+    IEEE802154_TRX_STATE_TRX_OFF,   /**< the transceiver state if off */
+    IEEE802154_TRX_STATE_RX_ON,     /**< the transceiver is ready to receive/receiving packets */
+    IEEE802154_TRX_STATE_TX_ON,     /**< the transceiver is ready to transmit/transmitting a frame */
+} ieee802154_trx_state_t;
+
+/**
+ * @brief IEEE802.15.4 Radio HAL events
+ *
+ * To follow the IEEE802.15.4 convention, an event that responds to a Request
+ * is a confirmation (Confirm). Otherwise an Indication.
+ */
+typedef enum {
+    /**
+     * @brief the transceiver detected a valid SFD
+     *
+     * This event is present if radio has @ref IEEE802154_CAP_IRQ_RX_START cap.
+     */
+    IEEE802154_RADIO_INDICATION_RX_START,
+
+    /**
+     * @brief the transceiver received a frame and lies in the
+     *        internal framebuffer.
+     *
+     * This indication should be generated only if CRC is valid and the frame
+     * passes the address matching filter (this includes ACK and Beacon frames).
+     * The latter only applies if the radio is not in promiscuous mode.
+     *
+     * The transceiver or driver MUST handle the ACK reply if the Ack Request
+     * bit is set in the received frame and promiscuous mode is disabled.
+     *
+     * The transceiver is in @ref IEEE802154_TRX_STATE_RX_ON state when
+     * this funcion is called, but with framebuffer protection (either
+     * dynamic framebuffer protection or disabled RX). Thus, the frame
+     * won't be overwritten before calling the @ref ieee802154_radio_read
+     * function. However, @ref ieee802154_radio_read MUST be called in
+     * order to receive new packets. If there's no interest in the
+     * frame, the function can be called with a NULL buffer to drop
+     * the frame.
+     */
+    IEEE802154_RADIO_INDICATION_RX_DONE,
+
+    /**
+     * @brief the transceiver finished sending a frame or the
+     *        retransmission procedure
+     *
+     * This event is present if radio has @ref IEEE802154_CAP_IRQ_TX_DONE cap.
+     * The upper layer should immediately call @ref
+     * ieee802154_radio_ops::confirm_transmit when on this event.
+     */
+    IEEE802154_RADIO_CONFIRM_TX_DONE,
+    /**
+     * @brief the transceiver reports that the ACK timeout expired
+     *
+     * This event is present if radio has @ref IEEE802154_CAP_IRQ_ACK_TIMEOUT.
+     */
+    IEEE802154_RADIO_CONFIRM_ACK_TIMEOUT,
```

Comment:
I will remove this one.
The reason is because the `confirm_transmit` function is the one who gets the TX status from the radio (so this would be redundant with `TX_STATUS_NO_ACK`

Reply:
done